genres/service.js

```diff
@@ -0,0 +1,90 @@
+import logger from '../utils/logger.js';
+import GenreModel from './models/genre-model.js';
+
+export const fetchAllGenres = async () => {
+  try {
+    const genres = await GenreModel.find();
+    if (genres.length !== 0) {
+      return genres;
+    }
+  } catch (error) {
+    logger.error('Error during fetching genres from database Genre', error);
+    throw new Error('Error fetching genres from database');
+  }
+};
+
+export const fetchGenreById = async (id) => {
+  try {
+    const genre = await GenreModel.findById(id);
+    if (genre !== null) {
+      logger.info(`Genre with id ${id} is '${genre.name}'`);
+      return genre;
+    } else {
+      logger.info(`No genres found by id '${id}'`);
+      return null;
+    }
+  } catch (error) {
+    logger.error('Error find Genre in MongoDB', error);
+    throw new Error('Error fetching Genre in MongoDB', error);
+  }
```

Comment:
you need return null to get error in controller, because you have following checking 
```suggestion
  if (genre === null) {
      return res.status(404).json({ message: 'Genre not found' });
      }
```

Reply:
Put 'return null' in 'else' scope